server/protocols/webadmin/webadmin-data/src/main/java/org/apache/james/webadmin/routes/VacationRoutes.java

```diff
@@ -0,0 +1,199 @@
+                .toDate(updateTo(vacationDto.getToDate()))
+                .isEnabled(updateTo(vacationDto.getEnabled()))
+                .build();
+            vacationRepository.modifyVacation(accountId, vacationPatch)
+                .then(notificationRegistry.flush(accountId)).block();
+            return Responses.returnNoContent(response);
+        } catch (JsonExtractException e) {
+            throw ErrorResponder.builder()
+                .statusCode(HttpStatus.BAD_REQUEST_400)
+                .type(ErrorResponder.ErrorType.INVALID_ARGUMENT)
+                .message("Malformed JSON input")
+                .cause(e)
+                .haltError();
+        }
+    }
+
+    public String setVacation(Request request, Response response) {
+        testUserExists(request);
+        try {
+            AccountId accountId = AccountId.fromString(request.params(USER_NAME));
+            VacationDTO vacationDto = jsonExtractor.parse(request.body());
+            VacationPatch vacationPatch = VacationPatch.builder()
+                .subject(modifyTo(vacationDto.getSubject()))
+                .textBody(modifyTo(vacationDto.getTextBody()))
+                .htmlBody(modifyTo(vacationDto.getHtmlBody()))
+                .fromDate(modifyTo(vacationDto.getFromDate()))
+                .toDate(modifyTo(vacationDto.getToDate()))
+                .isEnabled(modifyTo(vacationDto.getEnabled()))
+                .build();
+            vacationRepository.modifyVacation(accountId, vacationPatch)
+                .then(notificationRegistry.flush(accountId)).block();
+            return Responses.returnNoContent(response);
+        } catch (JsonExtractException e) {
+            throw ErrorResponder.builder()
+                .statusCode(HttpStatus.BAD_REQUEST_400)
+                .type(ErrorResponder.ErrorType.INVALID_ARGUMENT)
+                .message("Malformed JSON input")
+                .cause(e)
+                .haltError();
+        }
+    }
+
+    public String deleteVacation(Request request, Response response) {
+        testUserExists(request);
+        AccountId accountId = AccountId.fromString(request.params(USER_NAME));
+        VacationPatch vacationPatch = VacationPatch.builder()
+            .isEnabled(false)
+            .subject(ValuePatch.remove())
+            .textBody(ValuePatch.remove())
+            .htmlBody(ValuePatch.remove())
+            .fromDate(ValuePatch.remove())
+            .toDate(ValuePatch.remove())
+            .build();
+        vacationRepository.modifyVacation(accountId, vacationPatch)
+            .then(notificationRegistry.flush(accountId)).block();
+        return Responses.returnNoContent(response);
+    }
+
+    private void testUserExists(Request request) {
+        boolean exists;
+        try {
+            exists = usersRepository.contains(Username.of(request.params(USER_NAME)));
+        } catch (UsersRepositoryException e) {
+            exists = false;
+        }
```

Comment:
Mutability spotted ;-)

Here this could easily be solved with one more method extraction.

Reply:
Sigh, ok ok.